server/audit_math/suite.py

```diff
@@ -1,718 +0,0 @@
-    bp_loser_votes = bp_stratum.vote_totals[loser]
-
-    V = cvr_winner_votes - cvr_loser_votes + bp_winner_votes - bp_loser_votes
-
-    # The election is tied
-    if V == 0:
-        return 1.0
-
-    lambda_lower = (
-        np.amax(
-            [
-                cvr_winner_votes - cvr_loser_votes - cvr_stratum.num_ballots,
-                V - (bp_winner_votes - bp_loser_votes + bp_stratum.num_ballots),
-            ]
-        )
-        / V
-    )
-    lambda_upper = (
-        np.amax(
-            [
-                cvr_winner_votes - cvr_loser_votes + cvr_stratum.num_ballots,
-                V - (bp_winner_votes - bp_loser_votes - bp_stratum.num_ballots),
-            ]
-        )
-        / V
-    )
-
-    sample = bravo.compute_cumulative_sample(bp_stratum.sample)
-    bp_sample_winner_votes = sample[winner]
-    bp_sample_loser_votes = sample[loser]
-
-    reported_margin = contest.candidates[winner] - contest.candidates[loser]
-
-    Wn = bp_sample_winner_votes
-    Ln = bp_sample_loser_votes
-    Un = bp_stratum.sample_size - bp_sample_winner_votes - bp_sample_loser_votes
-    assert Wn >= 0, f"{Wn, Ln, Un}"
-    assert Ln >= 0, f"{Wn, Ln, Un}"
-    assert Un >= 0, f"{Wn, Ln, Un}"
-
-    T2 = (
-        lambda delta: 2
-        * bp_stratum.sample_size
-        * np.log(1 + reported_margin * delta / (2 * bp_stratum.num_ballots * GAMMA))
-    )
-    modulus = (
-        lambda delta: 2 * Wn * np.log(1 + reported_margin * delta)
-        + 2 * Ln * np.log(1 + reported_margin * delta)
-        + 2 * Un * np.log(1 + 2 * reported_margin * delta)
-        + T2(delta)
-    )
-
-    while True:
-        test_lambdas = np.arange(lambda_lower, lambda_upper + stepsize, stepsize)
-        if len(test_lambdas) < 5:
-            stepsize = (lambda_upper + 1 - lambda_lower) / 5
-            test_lambdas = np.arange(lambda_lower, lambda_upper + stepsize, stepsize)
-
-        fisher_pvalues = np.empty_like(test_lambdas)
-        for i, test_lambda in enumerate(test_lambdas):
-            try:
-                pvalue1 = np.min(
-                    [
-                        1,
-                        cvr_stratum.compute_pvalue(
-                            reported_margin, winner, loser, test_lambda
-                        ),
-                    ]
-                )
-            except ValueError:
-                pvalue1 = 0
-
-            try:
-                pvalue2 = np.min(
-                    [
-                        1,
-                        bp_stratum.compute_pvalue(
-                            reported_margin, winner, loser, 1 - test_lambda
-                        ),
-                    ]
-                )
-            except ValueError as e:
-                # If the sprt throws an error, set its pvalue to 0.
-                # This is per the Stark code
-                if "Alternative" in str(e):
-                    pvalue2 = 1.0
-                else:
-                    pvalue2 = 0
```

Comment:
I couldn't find this logic in the original code. Where did you see it? Also curious why the two different cases here based on the error message

Reply:
Here's the corresponding code for the null error: https://github.com/pbstark/CORLA18/blob/master/code/sprt.py#L94

Reply:
For the alternative, it's roughly here: https://github.com/pbstark/CORLA18/blob/master/code/sprt.py#L60

Following the same logic as above, if the alternative isn't compatible with the sample, something is really wacky, so return a p-value of 1.

Reply:
Similar comment to above - should the p-value function itself make these decisions?